src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxButton.java

```diff
@@ -228,6 +230,17 @@ protected void doRendererPaint(final Graphics g, final ButtonModel buttonModel,
 
         rendererPane.paintComponent(g, c, this, left, top, cWidth, height, shouldValidate); // h - (insets.top + insets.bottom) );
 
+        // fix for 8283214
+        // set the accessible name to the displayed text in JComboBox.
+        // screen magnifier queries to get the accessible name to display magnified text.
+
+        AccessibleContext ac = this.getAccessibleContext();
+        if (ac != null && !editable && c instanceof JLabel label) {
+            if (label.getText() != null) {
+                ac.setAccessibleName(label.getText());
+            }
+        }
```

Comment:
How it will work in other L&Fs?

Reply:
In case of other LAFs except **Aqua** in MacOS, the accessible component invoked for screen magnifier is an instance of `AccessibleJComboBox` and subsequently the a11y API called on `AccessibleJComboBox` to return the accessible name.

But for aqua, the accessible component invoked for screen magnifier is an instance for `AquaComboBoxButton`, so it is required to return the accessible button instance for a11y APIs to get called for retrieving accessible name.

Reply:
That's exactly my concern: in other L&Fs `AccessibleJComboBox` handles the scenario yet in Aqua L&F it is handled by a different component which is not a problem okay per se. What is a problem is that retrieving the accessible name depends on paint:
* if `AquaComboBoxButton.paint` hasn't been called for whatever reason, there's no accessible name;
* if the value in the combo box is changed, the accessible name isn't updated until paint occurs.

Reply:
> That's exactly my concern: in other L&Fs `AccessibleJComboBox` handles the scenario yet in Aqua L&F it is handled by a different component which is not a problem okay per se. What is a problem is that retrieving the accessible name depends on paint:
> 
>     * if `AquaComboBoxButton.paint` hasn't been called for whatever reason, there's no accessible name;
> 
>     * if the value in the combo box is changed, the accessible name isn't updated until paint occurs.

I rechecked again but couldn't find any other way to either set accessible name or get the current selected item other than `AquaComboBoxButton.paint`.

Reply:
>But for aqua, the accessible component invoked for screen magnifier is an instance for AquaComboBoxButton, so >it is required to return the accessible button instance for a11y APIs to get called for retrieving accessible name.

Can we inject some code into the AquaComboBoxButton so it will know what name should be used if that button is used "inside" combobox?

Reply:
> Can we inject some code into the AquaComboBoxButton so it will know what name should be used if that button is used "inside" combobox?

I didn't get you.

Reply:
> Can we inject some code into the AquaComboBoxButton so it will know what name should be used if that button is used "inside" combobox?
If I understood this correctly, we are setting the accessible name for the button and that can be retrieved by `getAccessibleName `API for the button if it is required inside combobox .